### ISSUE TITLE
Current app integrations

### ISSUE THREAD

**Issue by Maestro-Zacht** [Maestro-Zacht/aa-charlink#1]
Apps with integration code in Charlink:
- [x] AllianceAuth add character
- [x] [AllianceAuth CorpStats](https://allianceauth.readthedocs.io/en/latest/features/apps/corpstats.html)
- [x] [CorpTools](https://github.com/Solar-Helix-Independent-Transport/allianceauth-corp-tools)
- [x] [MemberAudit](https://gitlab.com/ErikKalkoken/aa-memberaudit)
- [x] [MiningTaxes](https://gitlab.com/arctiru/aa-miningtaxes)
- [x] [Structures](https://gitlab.com/ErikKalkoken/aa-structures)
- [x] [Moon Tools](https://gitlab.com/colcrunch/aa-moonstuff) (Deprecated)
- [x] [Moon Mining](https://gitlab.com/ErikKalkoken/aa-moonmining)
- [x] [CorpStats-Two](https://github.com/Solar-Helix-Independent-Transport/allianceauth-corpstats-two) (Deprecated)
- [x] [AFAT](https://github.com/ppfeufer/allianceauth-afat)
- [x] [Market Manager](https://gitlab.com/tactical-supremacy/aa-market-manager)
- [x] [AA Contacts](https://github.com/Maestro-Zacht/aa-contacts)
- [ ] [Standing Requests](https://gitlab.com/ErikKalkoken/aa-standingsrequests)

Apps with integration via hook:
- [x] [AA-Ledger](https://github.com/Geuthur/aa-ledger)
- [x] [AA-Skillfarm](https://github.com/Geuthur/aa-skillfarm)
- [x] [AA Contacts](https://github.com/Maestro-Zacht/aa-contacts)
- [x] [AA Metenox](https://gitlab.com/r0kym/aa-metenox)
- [x] [AA Dens](https://gitlab.com/r0kym/aa-dens)

**Comment by cpjet64** [comment]
Some modules that would be nice to have added:
Structures https://apps.allianceauth.org/apps/detail/aa-structures
Moon Tools https://apps.allianceauth.org/apps/detail/aa-moonstuff
Moon Mining https://apps.allianceauth.org/apps/detail/aa-moonmining
Fittings https://apps.allianceauth.org/apps/detail/fittings

**Comment by ppfeufer** [comment]
AFAT » https://github.com/ppfeufer/allianceauth-afat

**Comment by Maestro-Zacht** [comment]
> AFAT » https://github.com/ppfeufer/allianceauth-afat

I just checked your app and it doesn't seem it has a login functionality. It requests a token when adding esi fleets but there is no associated action such as registering the user or the token in a model, like other apps do. In addition to that if there is a valid token the user would need to select that anyway when adding an esi FAT, it would just skip the ccp login.

If I'm missing something pls let me know.

**Comment by ppfeufer** [comment]
Nah, that's pretty much it. Skipping the login and landing directly at the character selection would be the improvement/benefit here. Pretty much "pre-loading" the scopes.

Tokens are requested when:

- An ESI fleet is created (`esi-fleets.read_fleet.v1`)
- A user clicks on a FAT link to register their participation (`esi-location.read_location.v1`, `esi-location.read_ship_type.v1`, `esi-location.read_online.v1`)

**Comment by davidrobbins1** [comment]
Corporation Audit. I believe it is part of corp tools https://website.com/audit/corp/r/#/

**Comment by Maestro-Zacht** [comment]
Yes it is and i totally missed that. It'll be a little more difficult to implement though, because it require changing how the dynamic import works since atm it accepts 1 entry for each app. I'll do that after completing tests and AFAT integration.

**Comment by GurkeTonic** [comment]
It would be great to integrate the following module:
- aa-market-manager https://gitlab.com/tactical-supremacy/aa-market-manager

**Comment by Sutima** [comment]
Would be lovely to add https://gitlab.com/ErikKalkoken/aa-standingsrequests

**Comment by Maestro-Zacht** [comment]
I assume you'd like to add the login when you request standings to a corporation/alliance, and not the one for setting up the alliance. In that case, it's not possible at the moment since the token you add has different scopes based on the state the user is in, while only a static list of scopes can be set on charlink. This may change in the future, I'll leave the entry in the list.

**Comment by Geuthur** [comment]
- aa-ledger: [https://github.com/Geuthur/aa-ledger](https://github.com/Geuthur/aa-ledger)
- aa-skillfarm: [https://github.com/Geuthur/aa-skillfarm](https://github.com/Geuthur/aa-skillfarm)

I have already added the hook system to the ledger.

**Comment by r0kym** [comment]
Now integrated in [aa-metenox](https://gitlab.com/r0kym/aa-metenox) using an auth hook

Is also in [aa-dens](https://gitlab.com/r0kym/aa-dens) that should be released soonTM
